docs/configuration.md

```diff
@@ -90,83 +90,97 @@ Settings like this are typically be provided by the language-specific LSP client
-This is handled by the [hie-bios](https://github.com/mpickering/hie-bios) project.
+This is handled under the hood by the [hie-bios](https://github.com/mpickering/hie-bios) application.
+In turn, `hie-bios` needs some configuration to identify all files, GHC options, etc., needed to compile a project.
 
-**For a full explanation of how `hie-bios` determines the project build configuration, and how to configure it manually, refer to the [hie-bios README](https://github.com/mpickering/hie-bios/blob/master/README.md).**
+There are several ways to provide this configuration to `hie-bios`, detailed below.
 
-At the moment, `haskell-language-server` has support to automatically detect your project build configuration to handle most use cases.
+### Implicit configuration
+If no `hie.yaml` file is present, `haskell-language-server` automatically detects your `hie-bios` configuration.
+**For most cases, this works just fine, and is the recommended way.**
 
-*So using a explicit `hie.yaml` file will not likely fix your ide setup*. It will do it almost only if you see an error like `Multi Cradle: No prefixes matched`
+See [the hie-bios documentation](https://github.com/haskell/hie-bios#implicit-configuration) for more information on implicit configuration.
+
+### Explicit, generated configuration
+Maybe using the implicit configuration does not suit you.
+E.g., it does not work, or you prefer to have explicit configuration in your project.
+In that case, you can automatically generate a `hie.yaml` file, using [implicit-hie](https://github.com/Avi-D-coder/implicit-hie):
+
+```shell
+gen-hie > hie.yaml  # In the root directory of your project
+```
+
+### Explicit, manual configuration
+Maybe using the generated `hie.yaml` file does not suit you.
+E.g., it still does not work, or you want to fine-tune the configuration.
+We recommend to start from the `hie.yaml` file generated by `implicit-hie` (see previous section) and modify it to suit your needs.
 
-If the automatic detection fails with that error you can configure `hie-bios` using a `hie.yaml` file in the root of the workspace.
-A `hie.yaml` file **explicitly** describes how to setup the environment to compile the various parts of your project.
 For that you need to know what *components* your project has, and the path associated with each one.
 So you will need some knowledge about
 [stack](https://docs.haskellstack.org/en/stable/build_command/#components) or [cabal](https://cabal.readthedocs.io/en/latest/cabal-commands.html?#cabal-v2-build) components.
 
-You also can use [implicit-hie](https://github.com/Avi-D-coder/implicit-hie) to automatically generate `hie.yaml` files for
-the most common stack and cabal configurations
+**For a full explanation of how to configure it manually, refer to the [hie-bios documentation](https://github.com/mpickering/hie-bios/blob/master/README.md).**
+
+#### Examples of explicit `hie-yaml` configurations
```

Comment:
we should include the "empty cabal cradle" which mostly works for everything now:

```
cradle:
  cabal:
```

(you might ask "why doesn't implicit-hie generate this?" which is a good question...)

Reply:
Agreed, I added it in 63ba56a7d4bbcb0c9cfcaa08ac2968d6636cd1c1